### ISSUE TITLE
Textures dissapearing in Mickey's Wild Adventure

### ISSUE THREAD

**Issue by rapsaGnauJ** [libretro/beetle-psx-libretro#311]
The games run well except for this. Whenever Mickey gain height, a layer of textures (usually the floor and some more) dissapear.
![mickey s wild adventure-180303-121327](https://user-images.githubusercontent.com/24842781/36933617-91ff8af4-1edc-11e8-8923-f18a91d2044c.png)
![mickey s wild adventure-180303-121343](https://user-images.githubusercontent.com/24842781/36933618-921716d8-1edc-11e8-924e-37def796d943.png)
![mickey s wild adventure-180303-121358](https://user-images.githubusercontent.com/24842781/36933619-9231e102-1edc-11e8-815d-65f591e80402.png)
![mickey s wild adventure-180303-121405](https://user-images.githubusercontent.com/24842781/36933620-924a64c0-1edc-11e8-910f-711a4db59c34.png)
![mickey s wild adventure-180303-121438](https://user-images.githubusercontent.com/24842781/36933621-9264a6aa-1edc-11e8-8895-37730a114d3f.png)
![mickey s wild adventure-180303-121456](https://user-images.githubusercontent.com/24842781/36933622-927fc25a-1edc-11e8-9d69-60db4b0650a9.png)

**Comment by saftle** [comment]
Do you have any of the PGXP options enabled?

**Comment by rapsaGnauJ** [comment]
I first tried with it disabled but enabling them doesn't seem to help at all.

**Comment by rapsaGnauJ** [comment]
No, I haven't. Just tested with the retroarch core. I can't test it right now but I will as soon as I can. If you want to test it yourself, you can just get into the first level (as seen in the screenshots) and jump.

**Comment by rapsaGnauJ** [comment]
I've tested it right now and the game seems to work as it should (no textures dissapearing) in standalone mednafen. Tested in version 0.9.48.

**Comment by saftle** [comment]
Ah ok great. Are you using SW or HW and if you're using HW which renderer are you using (Software, GL, or Vulkan). Also what is your retroarch video driver set to?

**Comment by rapsaGnauJ** [comment]
I am using HW with vulkan render option and my retroarch video driver is set to gl. I'm such a noob so I don't know if these options are great or what...

**Comment by saftle** [comment]
If retroarch's driver is set to GL, then the HW renderer uses GL by default regardless of whether Vulkan is set or not. Same thing goes for if you have Vulkan as your driver, and GL set as the renderer, it'll auto-use vulkan. So the bug might only affect GL in this case, since you're using GL. I'd try changing the driver to Vulkan as well to see if the bug persists there.

**Comment by hizzlekizzle** [comment]
It would also be good to know if it happens with the software renderer.

**Comment by rapsaGnauJ** [comment]
I changed the video driver to vulkan and it looks nice for now! Thank you!
I also tested it with the SW core and it seems to work nice as well. Just a GL problem I guess.

**Comment by logos88** [comment]
I can confirm this bug with HW and GL in Linux

**Comment by i30817** [comment]
This is still happening.
